TACC/Core-Portal
# FP-635: Remove Font Awesome Icons

## To Do

- [x] [Confirm error message style and layout.](https://github.com/TACC/Core-Portal/pull/469#discussion_r697504347)



## Overview: ##

- Do **not** use Font Awesome icons (3rd party).
- **Do** use Cortal icons (in house).



## Related Jira tickets: ##

* [FP-635](https://jira.tacc.utexas.edu/browse/FP-635)



## Summary of Changes: ##

- __Remove Font Awesome dependencies.__
- Add & Update relevant test cases.
- __Replace icons:__

	<details>

	- Data Files checkbox layered icons
	  → new Checkbox
	- Data Files move modal back button
	  → faked Cortal icon
	- Data Files Project description submit error state with
	  → error Message left of button
	- Ticket Modal Reply form button error state with
	  → error Message left of button
	- Ticket Modal Reply Listing error state
	  → error Message

	</details>

- __Add **new** `<Checkbox>` component.__
- Add _basic_ support for layered icons (an "icon set").
- Add _faked_ Cortal icon for navigation arrow heads.
- __Remove template load of Font Awesome.__



## Testing Steps & UI Photos: ##

1. Data Files checkbox layered icons:

	<details>

	1. unchecked state is a black box: `icon-boxed`
	    <img width="400px" alt="GH-635 Data Files Checkbox - Checked State" src="https://user-images.githubusercontent.com/62723358/131025726-4cf38481-734f-4e76-8235-924a722ad6a8.png" />
	2. checked state is a black box, purple background, white check: `icon-approved-boxed-reverse`
	    <img width="400px" alt="GH-635 Data Files Checkbox - Default State" src="https://user-images.githubusercontent.com/62723358/131025729-1a69015f-c1b6-4992-be2e-800612c30897.png" />
	3. table header checkbox (select all) _still works_
	    <img width="400px" alt="GH-635 Data Files Checkbox - Select All" src="https://user-images.githubusercontent.com/62723358/131025730-3fca9958-d38c-4216-acb6-7df683200b94.png" />
	4. unchecking rows after selecting all via table header checkbox _still works_
	    <img width="400px" alt="GH-635 Data Files Checkbox - Uncheck Some" src="https://user-images.githubusercontent.com/62723358/131025732-6136503e-3fcb-4230-9df9-50b32ac031a6.png" />

	</details>

2. Data Files move modal back button:

	<details>

	1. select file (or some folder **not** `.APP_DATA`)
	2. hit move button
	3. navigate into a folder
	4. see back button has leftward pointing icon before it that looks like [design](https://xd.adobe.com/view/db2660cc-1011-4f26-5d31-019ce87c1fe8-ad17/screen/85634389-b55c-4166-9ccd-521d56777522)

	<img width="400px" alt="GH-635 Data Files Move Modal - Back Button" src="https://user-images.githubusercontent.com/62723358/131025837-9f0523c3-884a-4b15-8c51-d433ad79da4f.png" />

	</details>

3. Data Files Project (Shared Workspace) description submit error state:

	<details>

	1. confirm button is aligned to right
	2. cause error in submitting workspace desc edit
	    (to fake error, I changed [`updatingError`](https://github.com/TACC/Core-Portal/blob/22492d2/client/src/components/DataFiles/DataFilesModals/DataFilesProjectEditDescriptionModal.js#L111) to `true`)
	3. button should **not** have (!) icon
	4. error message should appear to left of button
	5. there should be space between message and button

	<img width="400px" alt="GH-635 Project Desc Edit - Submit Error" src="https://user-images.githubusercontent.com/62723358/131026231-87e8984d-dfda-4bc3-9365-63dcc44dfde1.png" />

	</details>

4. Ticket Modal Reply form button error state:

	<details>

	1. confirm button is aligned to right
	2. cause error in submitting reply
	    (to fake error, I changed [`replyingError`](https://github.com/TACC/Core-Portal/blob/5d3f9a3/client/src/components/Tickets/TicketModal.js#L100) to `true`)
	3. button should **not** have (!) icon
	4. error message should appear to left of button
	5. there should be space between message and button

	<img width="400px" alt="GH-635 Ticket Reply - Submit Error" src="https://user-images.githubusercontent.com/62723358/131026303-e8edec3e-f00e-4f80-a8d0-b26628eb49f8.png" />

	</details>

5. Ticket Modal Reply Listing error state:

	<details>

	<img width="400px" alt="GH-635 Ticket List - Load Progress" src="https://user-images.githubusercontent.com/62723358/131026337-cdfc3d26-48f4-4d5d-90ab-85ecf41143e9.png" />

	1. confirm loading spinner is centered in replies section (left panel)
	2. cause error in loading replies
	    (to fake error, I changed [`loadingError`](https://github.com/TACC/Core-Portal/blob/57fe446/client/src/components/Tickets/TicketModal.js#L202) to `true && !loading` __and__ commented out the [`history.map` code block](https://github.com/TACC/Core-Portal/blob/57fe446/client/src/components/Tickets/TicketModal.js#L207-L216))
	3. section should **not** have giant (!) icon
	4. error message should appear centered in replies section (left panel)

	<img width="400px" alt="GH-635 Ticket List - Load Error" src="https://user-images.githubusercontent.com/62723358/131026367-0fd96cd0-f111-41e3-8d56-e624d9d6ac7b.png" />

	</details>



## Notes: ##

See comments in new code.
